tests/formatter.rs

```diff
@@ -837,14 +844,16 @@ bar = { version = "0.1.0", optional = true }
 error: unused optional dependency
   |
 4 |    bar = { version = "0.1.0", optional = true }
-  |   __________________________________________^
-  |   _________^
-  |                               --------------- info: This should also be long but not too long
+  |   _________^__________________--------------^
+  |  |         |                  |
+  |  |_________|                  info: This should also be long but not too long
+  | ||
 5 | || this is another line
 6 | || so is this
 7 | || bar = { version = "0.1.0", optional = true }
-  | ||__________________________________________^ I need this to be really long so I can test overlaps
-  | ||_________________________^ I need this to be really long so I can test overlaps
+  | ||_________________________^________________^ I need this to be really long so I can test overlaps
+  | |__________________________|
+  |                            I need this to be really long so I can test overlaps
   |
```

Comment:
Is the diff being funky or is this really confusing to see the spans highlighted?

Reply:
Maybe I have the curse of knowledge, but it makes sense to me; what part is confusing?

Reply:
The diff makes it harder to read, but the new output matches rustc's